kernel/init.c

```diff
@@ -237,6 +242,10 @@ static void bg_thread_main(void *unused1, void *unused2, void *unused3)
 
 	z_sys_post_kernel = true;
 
+#ifdef CONFIG_MAIN_THREAD_WORK_QUEUE
+	k_queue_init(&k_main_work_q.queue);
+#endif
+
 	z_sys_device_do_config_level(_SYS_INIT_LEVEL_POST_KERNEL);
```

Comment:
I suppose that `z_sys_device_do_config_level` initialize default system workqueue thread here with its own stack, what if we want to have system workqueue that runs on main's stack ?